### DISCUSSION TITLE
How to use non-default user for replication

### DISCUSSION THREAD

**Discussion by sheyt0** [apache/kvrocks#2274] [Q&A] [answered]
Is it possible to use a non-default user (RO) for replication?

Something like
```
user *ro_user* on +@read +PSYNC +REPLCONF ~* >*ro_user_pass*
```
and
```
masteruser *ro_user*
masterauth *ro_user_pass*
```
in Redis

**Comment by git-hulk** [comment]
@sheyt0 Kvrocks doesn't support the ACL user for now, but you can use the namespace mechanism to separate the default user(admin) and other users, please refer: https://kvrocks.apache.org/docs/namespace

**Reply by sheyt0** [reply]
@git-hulk, thanks.
I'm using namespaces + requerepass on master + masterauth on replicas.
There's nothing stopping someone from taking masterauth
 value from config and use it to connect via redis-cli to master and make any changes.

**Reply by git-hulk** [reply, marked as answer]
@sheyt0 Yes, no way to workaround this without the ACL user feature.

**Reply by sheyt0** [reply]
Are there any plans to add support in the near future?

**Reply by git-hulk** [reply]
I have added it to our 2024 yearly plan: https://github.com/apache/kvrocks/discussions/1974, but  cannot promise the finished time.
